### DISCUSSION TITLE
Berry mqtt.publish fulltopic

### DISCUSSION THREAD

**Discussion by jeroenst** [arendst/Tasmota#18294] [Ideas]
Currently it's possible to publish to mqtt using a full specified topic.

I would like to have an option to publish to the topic configured at fulltopic so the mqtt fulltopic is not hardcoded in the berry script.

Example:

mqtt.publishft(prefix, payload, retain, start, len)

**Comment by barbudor** [comment]
Solution has already been provided in the past
See @sfromis example here:
https://github.com/arendst/Tasmota/discussions/17198#discussioncomment-4246710

Although a dedicated command could also make sense. PR welcomed.

**Reply by sfromis** [reply]
The example of `mqtt.publishft(prefix, payload, retain, start, len)` would not work anyway, as it does not provide the 3rd part of the topic. If you have a functional syntax, it would be very simple to create a Berry function using FullTopic to transform that and do a regular publish, along the lines of the FullTopic line in the code example linked by barbudor.

**Comment by jeroenst** [comment]
mqtt.publishft(prefix, subtopic, payload, retain, start, len) would do the trick.

**Reply by sfromis** [reply]
And then you can easily "create a Berry function using FullTopic to transform that and do a regular publish, along the lines of the FullTopic line in the code example linked by barbudor"

**Reply by jeroenst** [reply]
Maybe adding a command like mqqt.getfulltopic() would be a better option.
